rcl/test/rcl/test_get_node_names.cpp

```diff
@@ -162,3 +163,134 @@ TEST_F(CLASSNAME(TestGetNodeNames, RMW_IMPLEMENTATION), test_rcl_get_node_names)
+  });
+  rcl_context_t context = rcl_get_zero_initialized_context();
+  const char * security_context_name = "/default";
+  const char * argv[] = {"--ros-args", "--security-context", security_context_name};
+  ret = rcl_init(3, argv, &init_options, &context);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_shutdown(&context)) << rcl_get_error_string().str;
+    EXPECT_EQ(RCL_RET_OK, rcl_context_fini(&context)) << rcl_get_error_string().str;
+  });
+  std::multiset<std::tuple<std::string, std::string, std::string>>
+  expected_nodes, discovered_nodes;
+
+  rcl_node_t node1 = rcl_get_zero_initialized_node();
+  const char * node1_name = "node1";
+  const char * node1_namespace = "/";
+  rcl_node_options_t node1_options = rcl_node_get_default_options();
+  ret = rcl_node_init(&node1, node1_name, node1_namespace, &context, &node1_options);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  expected_nodes.insert(
+    std::make_tuple(
+      std::string(node1_name),
+      std::string(node1_namespace),
+      std::string(security_context_name)));
+
+  rcl_node_t node2 = rcl_get_zero_initialized_node();
+  const char * node2_name = "node2";
+  const char * node2_namespace = "/";
+  rcl_node_options_t node2_options = rcl_node_get_default_options();
+  ret = rcl_node_init(&node2, node2_name, node2_namespace, &context, &node2_options);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  expected_nodes.insert(
+    std::make_tuple(
+      std::string(node2_name),
+      std::string(node2_namespace),
+      std::string(security_context_name)));
+
+  rcl_node_t node3 = rcl_get_zero_initialized_node();
+  const char * node3_name = "node3";
+  const char * node3_namespace = "/ns";
+  rcl_node_options_t node3_options = rcl_node_get_default_options();
+  ret = rcl_node_init(&node3, node3_name, node3_namespace, &context, &node3_options);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  expected_nodes.insert(
+    std::make_tuple(
+      std::string(node3_name),
+      std::string(node3_namespace),
+      std::string(security_context_name)));
+
+  rcl_node_t node4 = rcl_get_zero_initialized_node();
+  const char * node4_name = "node2";
+  const char * node4_namespace = "/ns/ns";
+  rcl_node_options_t node4_options = rcl_node_get_default_options();
+  ret = rcl_node_init(&node4, node4_name, node4_namespace, &context, &node4_options);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  expected_nodes.insert(
+    std::make_tuple(
+      std::string(node4_name),
+      std::string(node4_namespace),
+      std::string(security_context_name)));
+
+  rcl_node_t node5 = rcl_get_zero_initialized_node();
+  const char * node5_name = "node1";
+  const char * node5_namespace = "/";
+  rcl_node_options_t node5_options = rcl_node_get_default_options();
+  ret = rcl_node_init(&node5, node5_name, node5_namespace, &context, &node5_options);
```

Comment:
@ivanpauno hmm, is this checking that the feature works with duplicated FQN/security context tuples?

Reply:
yes